aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -162,7 +167,21 @@ private String cleanWarehousePath(String path) {
 
   @Override
   protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
-    return new GlueTableOperations(glue, lockManager, catalogName, awsProperties, fileIO, tableIdentifier);
+    if (catalogProperties != null) {
+      S3FileIO s3FileIO = new S3FileIO();
```

Comment:
This does not necessarily need to be S3FileIO, I think we can still initialize the property normally by calling `initializeFileIO` with the updated catalog properties.